main.cpp

```diff
@@ -279,21 +203,79 @@ struct HeapA
  */
 
 #include <iostream>
+#include <cmath>
+
+struct FloatType;
+struct DoubleType;
+struct IntType;
+
+// FLOAT DECLARATION
```

Comment:
all include statements should go at the top of the source code
You already forward declared these types for the Point class. no need to forward declare them again.
- [ ] revise